### ISSUE TITLE
Clicking "Show all results" from autocomplete searchbar causes Bad Gateway error

### ISSUE THREAD

**Issue by adrienjoly** [openwhyd/openwhyd#213]
**Describe the bug**

**To Reproduce**
1. Go to openwhyd.org
2. Click on search bar
3. Tap some random text and wait for autocomplete results to appear
4. Click "Show all results..." => bad getaway error page from cloudflare

![image](https://user-images.githubusercontent.com/531781/62939239-1afb4480-bdd1-11e9-95d1-a8a43d2765a5.png)

**Expected behavior**

It should go to the search results page, like https://openwhyd.org/search?q=hellojd (for instance)

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Firefox

(also reported by Florent Pietot)

**Comment by IneoO** [comment]
```
 ===
web_1    | Tue, 24 Sep 2019 18:22:23 GMT, Error (concise trace)
web_1    |     at conciseTrace (/usr/src/app/app.js:40:10)
web_1    |     at Console.error (/usr/src/app/app.js:56:9)
web_1    | (node:16) UnhandledPromiseRejectionWarning: AlgoliaSearchError: Method not allowed with this API key
web_1    |     at success (/usr/src/app/node_modules/algoliasearch/src/AlgoliaSearchCore.js:375:32)
web_1    |     at <anonymous>
web_1    |     at process._tickCallback (internal/process/next_tick.js:188:7)
web_1    | ===
web_1    | Tue, 24 Sep 2019 18:22:23 GMT, Error (concise trace)
web_1    |     at conciseTrace (/usr/src/app/app.js:40:10)
web_1    |     at Console.error (/usr/src/app/app.js:56:9)
web_1    | (node:16) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
web_1    | ===
web_1    | Tue, 24 Sep 2019 18:22:23 GMT, Error (concise trace)
web_1    |     at conciseTrace (/usr/src/app/app.js:40:10)
web_1    |     at Console.error (/usr/src/app/app.js:56:9)
web_1    | (node:16) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
web_1    | ===
web_1    | Tue, 24 Sep 2019 18:22:23 GMT, Error (concise trace)
web_1    |     at conciseTrace (/usr/src/app/app.js:40:10)
web_1    |     at Console.error (/usr/src/app/app.js:56:9)
web_1    | (node:16) UnhandledPromiseRejectionWarning: AlgoliaSearchError: Method not allowed with this API key
web_1    |     at success (/usr/src/app/node_modules/algoliasearch/src/AlgoliaSearchCore.js:375:32)
web_1    |     at <anonymous>
web_1    |     at process._tickCallback (internal/process/next_tick.js:188:7)
web_1    | ===
web_1    | Tue, 24 Sep 2019 18:22:23 GMT, Error (concise trace)
web_1    |     at conciseTrace (/usr/src/app/app.js:40:10)
web_1    |     at Console.error (/usr/src/app/app.js:56:9)
web_1    | (node:16) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
web_1    | ===
web_1    | Tue, 24 Sep 2019 18:22:23 GMT, Error (concise trace)
web_1    |     at conciseTrace (/usr/src/app/app.js:40:10)
web_1    |     at Console.error (/usr/src/app/app.js:56:9)
web_1    | (node:16) UnhandledPromiseRejectionWarning: AlgoliaSearchError: Method not allowed with this API key
web_1    |     at success (/usr/src/app/node_modules/algoliasearch/src/AlgoliaSearchCore.js:375:32)
web_1    |     at <anonymous>
web_1    |     at process._tickCallback (internal/process/next_tick.js:188:7)
web_1    | ===
web_1    | Tue, 24 Sep 2019 18:22:23 GMT, Error (concise trace)
web_1    |     at conciseTrace (/usr/src/app/app.js:40:10)
web_1    |     at Console.error (/usr/src/app/app.js:56:9)
web_1    | (node:16) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 6)
web_1    | ===
web_1    | Tue, 24 Sep 2019 18:22:23 GMT, Error (concise trace)
web_1    |     at conciseTrace (/usr/src/app/app.js:40:10)
web_1    |     at Console.error (/usr/src/app/app.js:56:9)
web_1    | (node:16) UnhandledPromiseRejectionWarning: AlgoliaSearchError: Method not allowed with this API key
web_1    |     at success (/usr/src/app/node_modules/algoliasearch/src/AlgoliaSearchCore.js:375:32)
web_1    |     at <anonymous>
web_1    |     at process._tickCallback (internal/process/next_tick.js:188:7)
web_1    | ===
web_1    | Tue, 24 Sep 2019 18:22:23 GMT, Error (concise trace)
web_1    |     at conciseTrace (/usr/src/app/app.js:40:10)
web_1    |     at Console.error (/usr/src/app/app.js:56:9)
web_1    | (node:16) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 8)
```

**Comment by adrienjoly** [comment]
to reproduce, quickly search for "t", by pressing "t" and "enter", very quickly.

```
web_1    | === Tue, 24 Sep 2019 18:40:37 GMT GET /search?q=ttt&format=html&context=header&_=1569350395219 (search.controller) {"q":"ttt","format":"html","context":"header","_":"1569350395219"}
web_1    | search playlists { q: 'ttt' }
web_1    | => 0 hits
web_1    | search posts { q: 'ttt' }
web_1    | => 0 hits
web_1    | search tracks { q: 'ttt' }
web_1    | => 0 hits
web_1    | search users { q: 'ttt' }
web_1    | => 0 hits
web_1    | === Tue, 24 Sep 2019 18:40:37 GMT GET /search?q=t&format=html&context=header&_=1569350395220 (search.controller) {"q":"t","format":"html","context":"header","_":"1569350395220"}
web_1    | search playlists { q: 't' }
web_1    | => 2 hits
web_1    | search posts { q: 't' }
web_1    | => 1 hits
web_1    | search tracks { q: 't' }
web_1    | => 0 hits
web_1    | search users { q: 't' }
web_1    | => 1 hits
web_1    | === Tue, 24 Sep 2019 18:40:37 GMT GET /search?q=t&_=1569350395221 (search.controller) {"q":"t","_":"1569350395221"}
web_1    | search playlists { q: 't' }
web_1    | => 2 hits
web_1    | search posts { q: 't' }
web_1    | /usr/src/app/node_modules/mongodb/lib/utils.js:123
web_1    |     process.nextTick(function() { throw err; });
web_1    |                                   ^
web_1    | 
web_1    | TypeError: path must be absolute or specify root to res.sendFile
web_1    |     at ServerResponse.sendFile (/usr/src/app/node_modules/express/lib/response.js:425:11)
web_1    |     at renderFile (/usr/src/app/app/controllers/uploadedFile.js:133:14)
web_1    |     at renderImg (/usr/src/app/app/controllers/uploadedFile.js:151:12)
web_1    |     at /usr/src/app/app/controllers/uploadedFile.js:192:9
web_1    |     at /usr/src/app/app/models/post.js:213:5
web_1    |     at handleCallback (/usr/src/app/node_modules/mongodb/lib/utils.js:120:56)
web_1    |     at /usr/src/app/node_modules/mongodb/lib/collection.js:1416:5
web_1    |     at handleCallback (/usr/src/app/node_modules/mongodb/lib/utils.js:120:56)
web_1    |     at /usr/src/app/node_modules/mongodb/lib/cursor.js:683:5
web_1    |     at handleCallback (/usr/src/app/node_modules/mongodb-core/lib/cursor.js:171:5)
web_1    | npm ERR! code ELIFECYCLE
web_1    | npm ERR! errno 1
```

**Comment by adrienjoly** [comment]
Also, doing a " " search on openwhyd.org results in a never-ending loading animation + errors in js console + `content.bundle.js.map` user profile trying to be opened! (from server logs)

<img width="516" alt="Capture d’écran 2019-09-24 à 21 24 15" src="https://user-images.githubusercontent.com/531781/65543711-ffc43e80-df11-11e9-8086-c49d0376d782.png">

<img width="729" alt="Capture d’écran 2019-09-24 à 21 24 15 copie" src="https://user-images.githubusercontent.com/531781/65543716-02269880-df12-11e9-97d7-8367a40ac3a4.png">

**Comment by adrienjoly** [comment]
fixed by #225
